src/Storages/StorageDistributed.cpp

```diff
@@ -1031,6 +1031,57 @@ StoragePolicyPtr StorageDistributed::getStoragePolicy() const
     return storage_policy;
 }
 
+Cluster::Addresses StorageDistributed::parseAddresses(const std::string & name) const
+{
+    Cluster::Addresses addresses;
+
+    const auto & cluster = getCluster();
+    const auto & shards_info = cluster->getShardsInfo();
+    const auto & shards_addresses = cluster->getShardsAddresses();
+
+    for (auto it = boost::make_split_iterator(name, boost::first_finder(",")); it != decltype(it){}; ++it)
+    {
+        const std::string & dirname = boost::copy_range<std::string>(*it);
+        Cluster::Address address = Cluster::Address::fromFullString(dirname);
+        if (address.shard_index)
```

Comment:
Let's put the same comments as in https://github.com/ClickHouse/ClickHouse/blob/fac83efdd25541b42baa6e47b8b71d18f4f47867/src/Storages/Distributed/DistributedAsyncInsertDirectoryQueue.cpp#L265